sdk/core/core/inc/az_http_policy.h

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_HTTP_POLICY_H
+#define AZ_HTTP_POLICY_H
+
+#include <az_contract.h>
+#include <az_http_request_builder.h>
+#include <az_pair.h>
+#include <az_result.h>
+#include <az_span.h>
+#include <az_str.h>
+
+#include <_az_cfg_prefix.h>
+
+// PipelinePolicies
+//   Policies are non-allocating caveat the TransportPolicy
+//   Transport policies can only allocate if the transport layer they call allocates
+// Client ->
+//  ===HttpPipelinePolicies===
+//    UniqueRequestID
+//    Retry
+//    Authentication
+//    Logging
+//    Buffer Response
+//    Distributed Tracing
+//    TransportPolicy
+//  ===Transport Layer===
+
+typedef struct az_http_policy az_http_policy;
+
+// PipelinePolicies must implement the process function
+//
+typedef AZ_NODISCARD az_result (*az_http_policy_pfnc_process)(
+    az_http_policy * policies,
+    az_http_request_builder * hrb,
+    az_mut_span const * const response);
+
+AZ_NODISCARD az_result az_http_pipeline_policy_uniquerequestid(
+    az_http_policy * const policies,
+    az_http_request_builder * const hrb,
+    az_mut_span const * const response);
```

Comment:
sorry for nit comments, but please insert newline between multi-line  function declarations

Reply:
Hmmm, too bad Clang doesn't catch that.  Fixed